src/frontend/js/pages/TeacherDashboardContractsLoader/TeacherDashboardContracts/index.tsx

```diff
@@ -0,0 +1,147 @@
+import { defineMessages, useIntl } from 'react-intl';
+
+import { Button, DataGrid, SortModel, usePagination } from '@openfun/cunningham-react';
+import { useEffect, useMemo, useState } from 'react';
+import { useContracts } from 'hooks/useContacts';
+import Banner, { BannerType } from 'components/Banner';
+import { PER_PAGE } from 'settings';
+import { ContractFilters } from 'types/Joanie';
+
+const messages = defineMessages({
+  errorNoCourseProductRelation: {
+    defaultMessage: "This product doesn't exist",
+    description: 'Message displayed when requested course product relation is not found',
+    id: 'components.TeacherDashboardContracts.errorNoCourseProductRelation',
+  },
+});
+
+export interface TeacherDashboardContractsProps {
+  courseId?: string;
+  organizationId?: string;
+  page?: string;
+}
+
+const TeacherDashboardContracts = ({
+  courseId,
+  organizationId,
+  page,
+}: TeacherDashboardContractsProps) => {
+  const intl = useIntl();
+  const [contractFilters, setContractFilters] = useState<ContractFilters>({
+    course_id: courseId,
+    organization_id: organizationId,
+    page: page ? parseInt(page, 10) : 1,
+    page_size: PER_PAGE.teacherContractList,
+  });
+  useEffect(() => {
+    setContractFilters((currentFilters: ContractFilters) => ({
+      ...currentFilters,
+      organization_id: organizationId,
+    }));
+  }, [organizationId]);
+
+  const {
+    items: contracts,
+    meta,
+    states: { fetching, isFetched },
+  } = useContracts(contractFilters);
+
+  const [sortModel, setSortModel] = useState<SortModel>([
+    {
+      field: 'learnerName',
+      sort: 'desc',
+    },
+    {
+      field: 'productTitle',
+      sort: 'desc',
+    },
+    {
+      field: 'signDate',
+      sort: 'desc',
+    },
+  ]);
+
+  const pagination = usePagination({
+    defaultPage: page ? parseInt(page, 10) : 1,
+    pageSize: PER_PAGE.teacherContractList,
+  });
+  useEffect(() => {
+    if (isFetched) {
+      pagination.setPagesCount(Math.ceil(meta!.pagination!.count / PER_PAGE.teacherContractList));
```

Comment:
Why not use contracts.length? I understand that it's exactly the same thing, but as you use contracts in the dependencies, it seems easier to understand to use contract.length. Or in this case, change the dependency of the useEffect

Reply:
```
useEffect(() => {
    if (isFetched) {
      pagination.setPagesCount(Math.ceil(contracts.length / PER_PAGE.teacherContractList));
    }
  }, [contracts, isFetched]);
```

or 

```
useEffect(() => {
    if (isFetched) {
       pagination.setPagesCount(Math.ceil(meta!.pagination!.count / PER_PAGE.teacherContractList));
    }
  }, [meta.pagination.count, isFetched]);
```

Reply:
`contracts.lenght` will be 25 or less (the pagination page size).
Here we want the total number of page, this information is in `pagination.count`